monitor/monitor.go

```diff
@@ -111,12 +107,58 @@ func (m *Monitor) handleConnection(server net.Listener) {
 		}
 
 		mutex.Lock()
+		first := listeners.Len() == 0
 		listeners.PushBack(conn)
 		log.WithField("count.listener", listeners.Len()).Info("New monitor connected.")
+		/* Since this is our first listener that means the event loop
+		 * is not running and has to be started.
+		 */
+		if first {
```

Comment:
This could be `if first := listeners.Len() == 0; first {``` since we don't use first otherwise

Reply:
This does not do what you think it does. We only want to start the event loop if this is really our first listener. If you move the initialization further down after the list is being modified the meaning of `first` is not the same anymore.

Reply:
Oh! I see, yes. We also can't start the main loop since it would then exit because there are no listeners. You are totally right!